frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts

```diff
@@ -59,24 +65,73 @@ export const useResolvedExtensions: UseResolvedExtensions = require('@console/dy
+ * @param {React.ComponentType} [NoDataEmptyMsg] - (optional) no data empty message component
+ * @param {React.ComponentType} [EmptyMsg] - (optional) empty message component
+ * @param {function} [scrollNode] - (optional) function to handle scroll
+ * @param {string} [label] - (optional) label for table
+ * @param {string} [ariaLabel] - (optional) aria label
+ * @param {TableGridBreakpoint} [gridBreakPoint] - sizing of how to break up grid for responsiveness
+ * @param {function} [onSelect] - (optional) function for handling select of tbale
+ * @param {R} [rowData] - (optional) data specific to row
+ * @example
+ * ```ts
+ * const MachineList: React.FC<MachineListProps> = (props) => {
+ *   return (
+ *     <VirtualizedTable<MachineKind>
+ *      {...props}
+ *      aria-label='Machines'
+ *      columns={getMachineColumns}
+ *      Row={getMachineTableRow}
+ *     />
+ *   );
+ * }
+ * ```
+ */  
 export const VirtualizedTable: VirtualizedTableFC = require('@console/internal/components/factory/Table/VirtualizedTable')
   .default;
+
+/**
+ * Component for displaying table data within a table row
+ * @param {string} id - unique id for table
+ * @param {Set<string>} activeColumnIDs - active columns
+ * @param {string} [className] -  (optional) option class name for styling
+ * @example
+ * ```ts
+ * const PodRow: React.FC<RowProps<K8sResourceCommon>> = ({ obj, activeColumnIDs }) => {
+ *   return (
+ *     <>
+ *       <TableData id={columns[0].id} activeColumnIDs={activeColumnIDs}>
+ *         <ResourceLink kind="Pod" name={obj.metadata.name} namespace={obj.metadata.namespace} />
+ *       </TableData>
+ *       <TableData id={columns[1].id} activeColumnIDs={activeColumnIDs}>
+ *         <ResourceLink kind="Namespace" name={obj.metadata.namespace} />
+ *       </TableData>
+ *     </>
+ *   );
+ * };
+ * ```
+ */
 export const TableData: React.FC<TableDataProps> = require('@console/internal/components/factory/Table/VirtualizedTable')
   .TableData;
 
 /**
  * A hook that provides a list of user-selected active TableColumns.
  * @param options - Which are passed as a key-value map
- * @param options.columns - An array of all available TableColumns
- * @param options.showNamespaceOverride - (optional) If true, a namespace column will be included, regardless of column management selections
- * @param options.columnManagementID - (optional) A unique id used to persist and retrieve column management selections to and from user settings. Usually a 'group~verion~kind' string for a resource.
+ * @param {TableColumn<D>[]} options.columns - An array of all available TableColumns
+ * @param {boolean} [options.showNamespaceOverride] - (optional) If true, a namespace column will be included, regardless of column management selections
+ * @param {string} [options.columnManagementID] - (optional) A unique id used to persist and retrieve column management selections to and from user settings. Usually a 'group~verion~kind' string for a resource.
```

Comment:
@zherman0 data on this part seems to need to be updated
![image](https://user-images.githubusercontent.com/78589509/174937839-c2d441c5-3dbb-4796-8fb8-85ff2dfa1f5c.png)

Reply:
The documentation is correct but this is not rendered properly in the github markdown. Hopefully this can be fixed in the future.